refs/schemas/protocols/RFC-PRMP.md

```diff
@@ -0,0 +1,126 @@
+
+- The Systems CAN apply the Protocol in the context of different [Scenarios][Scenarios].
+
+- The Protocol can be used regardless of the Systems exchanging data in a one-off data dump, or through a continuous connection.
+
+- The network of Systems exchanging data related to a Data Subject CAN be cyclical. It MUST not be assumed to be a complete graph. It CAN also be highly volatile i.e. Systems can participate on an ad-hoc basis, only once, disappear and reappear.
+
+## Design Choices
+
+- The Protocol is agnostic from any particular way to address Systems and transfer messages to and from them. Systems are responsible of knowing how to identify and address another System with which they exchange data or any object defined in PRIV.
+
+- Systems process Privacy Requests and other PRIV objects (e.g. legal-base related events and consents) in order of their dates, and regardless of the order in which they have gain knowledge of them using the Protocol.
+
+- The Protocol is agnostic of the level of inter-system trust. A System MAY be implemented to trust other Systems to authenticate the Data Subject, covey responses, tell the truth or not. The security implications of such choices are out of scope of the Protocol.
+
+- The content of a message exchanged between two Systems is a PRIV object (e.g. [Privacy Request](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md#privacy-request), [Privacy Request Response](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md#privacy-request), [Consent](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md#consent))
+
+## Proposal
+
+### Multicast APIs
+
+To participate in exchange of PRIV objects, Systems expose an API. We call the System calling the API the **Calling System** and the System exposing the API the **Exposing System**. The API MAY have the following functions:
+
+#### push
+
+Updates the information of the Exposing System with novel information from the Calling System.
+
+#### pull
+
+Updates the information of the Calling System with novel information from the Exposing System.
+
+#### merge
+
+Equivalent to push and pull. Both Systems get updated with novel information from the other system.
+
+#### compare
+
+Tells the Calling System if the Exposing System has novel information with regards to what the Calling System already knows.
+
+## Design Implications for Systems Implementing PRMP
+
+### Underlying Data Structure
+
+Systems have the interest to store PRIV objects in a data structure equivalent to a hash map or to a message queue, capable of easy comparison and efficient compare function.
+
+### Callback Mechanism
+
+Systems MUST implement, internally, a callback mechanism, with a message queue to account for asynchronous nature of the Protocol.
```

Comment:
Can you give an example with the callback mechanism?

Reply:
No. This is just taken for the discussion you guys had, that Amaury told me about in a comment. I do understand what you have in mind, but I am note sure I can give an example.